lib/pharos/addon.rb

```diff
@@ -5,6 +5,20 @@
 require_relative 'logging'
 
 module Pharos
+  # @param name [String]
+  # @return [Pharos::Addon]
+  def self.addon(name, &block)
+    klass = Class.new(Pharos::Addon, &block).tap do |addon|
+      addon.addon_location = block.source_location.first
+      addon.name(name)
```

Comment:
This overrides whatever `name ...` is within the addon itself, making it meaningless... risks having the wrong `name ...` within the addon itself, making it misleading.

Reply:
So what is your suggestion? Try to make `.name` private or?

Reply:
I don't know, but repeating the name in both the `Pharos.addon '...'` and within `name '...'` doesn't make sense. Maybe keep the `Pharos.addon '...'` name and remove the set-form of `def name(name)`... maybe also remove the entire `def name` to fix #241 at the same time.

Reply:
Fixed.